### PR TITLE
Publish event members

### PR DESCRIPTION
**Pull request by lucaswerkmeister**

### CODE REVIEW

**Comment by gkoz** [comment]
It would be nice to have some safe solution to ensure that the event record we receive in the signal handler is of the correct type (probably implement `FromGlibPtr` for events).

**Comment by lucaswerkmeister** [comment]
I’m sorry, I don’t understand – is there some problem with this PR? Should some members remain unpublished?

**Comment by gkoz** [comment]
These belong in `gdk-sys` actually, as `gdk` shouldn't expose the foreign types obviously. So a proper approach is to move them there (publishing the fields) and then most likely create wrapper structs with accessor methods that would take care of any conversions.

**Comment by lucaswerkmeister** [comment]
Okay, but in the meantime, is there any reason why some events (e. g. `EventTouch`, `EventScroll`, `EventKey`) have public members, and others (e. g. `EventMotion`, `EventButton`) don’t? I assumed this was simply an oversight. AFAICT, there’s currently no way to get, for instance, the coordinates of a click event, and that seems… inconvenient :D

**Comment by gkoz** [comment]
Can't answer for why some of the stuff has been done the way it is but it's way unfinished for sure.
I guess there's no harm in publishing scalar types. Do you want raw pointers as well?

**Comment by lucaswerkmeister** [comment]
> Do you want raw pointers as well?

Not really… I’ll add a commit to unpublish them (including those that were public before).

**Comment by gkoz** [comment]
What, the event structs are missing `#[repr(C)]`? :/
Please add that and it will allow you to remove `allow(dead_code)`.

File: src/events.rs [hunk 1]
```diff
@@ -72,6 +72,8 @@ pub trait Event: Sized {
     }
 }
 
+// TODO unfisished, see some discussion in #21
```
Comment (gkoz) [inline src/events.rs]: Make this #25

Comment (lucaswerkmeister) [inline src/events.rs]: done (squash + rebase), also fixed the typo in “unfi*s*ished”

**Comment by lucaswerkmeister** [comment]
Done.

**Comment by gkoz** [comment]
:+1:

**Comment by GuillaumeGomez** [comment]
Here I am ! @lucaswerkmeister: Nice job and thanks for your work !

**Comment by lucaswerkmeister** [comment]
Cool, thanks for merging!

**Pull request merged**